cmd/packaging/windows-msi.go

```diff
@@ -229,27 +222,37 @@ func windowsMsiProcessFiles(path string) {
 		} else {
 			if len(relativePath) > 1 {
 				directoryRefsFileContent = append(directoryRefsFileContent,
-					fmt.Sprintf(`<DirectoryRef Id="FLUTTERASSETSDIRECTORY_%s">`, hashSha1(strings.Join(relativePath[:len(relativePath)-1], ""))),
+					fmt.Sprintf(`<DirectoryRef Id="APPLICATIONROOTDIRECTORY_%s">`, hashSha1(strings.Join(relativePath[:len(relativePath)-1], ""))),
 				)
 			} else {
 				directoryRefsFileContent = append(directoryRefsFileContent,
-					`<DirectoryRef Id="FLUTTERASSETSDIRECTORY">`,
+					`<DirectoryRef Id="APPLICATIONROOTDIRECTORY">`,
 				)
 			}
-			fileSource := filepath.Join("build", "flutter_assets", strings.Join(relativePath, pathSeparator))
+			fileSource := filepath.Join("build", strings.Join(relativePath, pathSeparator))
 			directoryRefsFileContent = append(directoryRefsFileContent,
-				fmt.Sprintf(`<Component Id="flutter_assets_%s" Guid="*">`, id),
-				fmt.Sprintf(`<File Id="flutter_assets_%s" Source="%s" KeyPath="yes"/>`, id, fileSource),
+				fmt.Sprintf(`<Component Id="build_%s" Guid="%s">`, id, genGUID()),
+				fmt.Sprintf(`<File Id="build_%s" Source="%s" KeyPath="yes"/>`, id, fileSource),
 				"</Component>",
 				"</DirectoryRef>",
 			)
 			componentRefsFileContent = append(componentRefsFileContent,
-				fmt.Sprintf(`<ComponentRef Id="flutter_assets_%s"/>`, id),
+				fmt.Sprintf(`<ComponentRef Id="build_%s"/>`, id),
 			)
 		}
 	}
 }
 
+func genGUID() string {
+	b := make([]byte, 16)
+	_, err := rand.Read(b)
+	if err != nil {
+		log.Errorf("Failed to generate GUID: %v", err)
+		os.Exit(1)
+	}
+	return strings.ToUpper(fmt.Sprintf("%x-%x-%x-%x-%x", b[0:4], b[4:6], b[6:8], b[8:10], b[10:]))
```

Comment:
I think a GUID is a bit more complex than just random bytes..
http://guid.one/guid/make
Perhaps we should use an existing package to generate a V4(?) GUID?

Reply:
https://github.com/google/uuid looks like a proper library for that